geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -936,6 +934,8 @@ public static GemFireCacheImpl getForPdx(String reason) {
         }
       }
     } // synchronized
+
+    clientMetadataService = new ClientMetadataService(this);
```

Comment:
Instead of creating a clientMetadataService for all caches, try to only do it for client caches. In order to also make the clientMetadataService final I did the following (the "+" lines are the ones I added):
+        this.clientMetadataService = null;
       } else {
         logger.info("Running in client mode");
         this.resourceEventsListener = null;
+        this.clientMetadataService = new ClientMetadataService(this);